src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,97 @@
+ */
+public class Storage {
+    private final String filePath;
+    /**
+     *  Constructor to create data storage object
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves data from duke.tasklist.TaskList and store into file specified by this.filePath
+     *
+     * @params list to be saved
+     */
+    public void saveData(TaskList taskList) throws DukeException {
+        try {
+            FileWriter fileWriter = new FileWriter(this.filePath);
+            for (int i = 0; i < taskList.getSize(); i++) {
+                Task task = taskList.getTask(i);
+                StringBuilder taskString = new StringBuilder();
+                taskString.append(task.getEventType());
+                taskString.append(" | " + task.getStatusIcon() + " | ");
+                taskString.append(task.getStorageDetails() + "\n");
+                fileWriter.write(taskString.toString());
+            }
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    /**
+     *  Load data from file path and stores it in an ArrayList for duke.Duke
+     *
+     * @return ArrayList containing tasks from file based on file path
+     */
+
+    public ArrayList<Task> loadData() throws DukeException {
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        File file = new File(this.filePath);
+        try {
+            // create file if file is empty
+            if (!file.exists()) {
+                file.getParentFile().mkdir();
+                file.createNewFile();
+            }
+            Scanner fileReader = new Scanner(file);
+            Task task;
+            while (fileReader.hasNextLine()) {
+                String data = fileReader.nextLine();
+                String[] dataArray = data.split("\\| ");
+                String eventType = dataArray[0].trim();
+                String isDone = dataArray[1].trim();
+                if (eventType.equals("T")) {
+                    task = new ToDo(dataArray[2].trim());
+                } else if (eventType.equals("D")) {
+                    LocalDate deadline = LocalDate.parse(dataArray[3].trim());
+                    task = new Deadline(dataArray[2].trim(), deadline);
+                } else {
+                    String[] eventDetails = dataArray[3].split("- ");
+                    LocalDate startDate = LocalDate.parse(eventDetails[0].trim());
+                    LocalDate endDate = LocalDate.parse(eventDetails[1].trim());
+                    task = new Event(dataArray[2].trim(), startDate, endDate);
+                }
+                if (isDone.equals("X")) {
+                    task.markAsDone();
+                }
+                taskList.add(task);
+            }
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+        return taskList;
+    }
```

Comment:
Maybe decreasing LOC could increase the readability of this method (more than 30 LOC)

Reply:
good idea! I should perhaps use some methods from the Parser class.